scheduler/server/task_runner.go

```diff
@@ -107,7 +107,6 @@ func (r *taskRunner) run() error {
 
 	// Update taskErr state if it's empty or if we're doing deadletter..
 	if taskErr.st.State == runner.UNKNOWN {
-		taskErr.st.State = runner.FAILED
```

Comment:
this is the actual fix.

Reply:
runner.UNKNOWN state might not be correctly mapped to a failing exit code for the client

Reply:
we tested the behavior in jbruno/devel (and rcouto/devel): started tryout job, killed one of the workers running a task and saw sickle report the task as failed and restarting.
sample output:
I] 15:02:55] Task Failure: src/go/twitter.biz/mgci/sq/validator:: (pants), (job ID: d775a460-ef42-4297-4504-8600ed23fa80, task prefix: -4), finished in 107s
I] 15:02:55]
I] 15:03:01] Retrying ./scoot/config/scripts/run_task/pants/pantsRun.sh ./pants --pants-config-files=pants.ini --pants-config-files=pants.ini.scoot --pants-supportdir=pants-support --no-enable-pantsd test lint src/go/twitter.biz/mgci/sq/validator::, (job ID: 50b2d799-90d3-414c-6edd-89ffdb68137b)
I] 15:03:01] Starting testing of 1 target(s). Job ID: 50b2d799-90d3-414c-6edd-89ffdb68137b

Reply:
yeah, only thing I can think of is that this could cause some odd downstream behavior. Sickle does seem to generally group this together with FAILED as "lost" though, and has a few tests referencing unknown.

Ultimately the comment in stateful scheduler around where we set flaky is leading me to believe the ultimate root cause is poor distinction around these states generally. But this might be enough to fix not suspending workers in this particular case